File: src/main/java/com/circleescape/server/ScoreboardAPI.java
package com.circleescape.server;

import com.circleescape.server.scoreboard.data.Score;
import com.circleescape.server.scoreboard.data.ScoreRepository;
import lombok.extern.log4j.Log4j2;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.bind.annotation.*;

import java.time.LocalDateTime;

@Log4j2
@CrossOrigin
@RestController
@Service
public class ScoreboardAPI {

	private final ScoreRepository scoreRepository;

	@Autowired
	public ScoreboardAPI(ScoreRepository scoreRepository) {
		this.scoreRepository = scoreRepository;
	}

	@RequestMapping("/")
	public String showRules() {
		return "Hi there";
	}

	@RequestMapping("/rules")
	public String showRules2() {
		return "Hi there";
	}



	@PostMapping("/scoreboard/{gameId}/postName")
	public ResponseEntity<Void> postNameOnScoreboard(
			@PathVariable long gameId,
			@io.swagger.v3.oas.annotations.parameters.RequestBody
			@RequestBody String name) {
		boolean gameFinished = true; //TODO

		LocalDateTime timestamp = LocalDateTime.now();

		Score scoreBE = new Score(gameId, name, 42, timestamp);

		scoreRepository.save(scoreBE);

		return ResponseEntity.ok().build();
	}

}

File: src/main/kotlin/com/circleescape/server/ScoreboardAPI.kt
package com.circleescape.server

import org.springframework.web.bind.annotation.CrossOrigin
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.http.ResponseEntity
import java.time.LocalDateTime
import com.circleescape.server.scoreboard.data.Score
import com.circleescape.server.scoreboard.data.ScoreRepository
import org.springframework.web.bind.annotation.PathVariable
import java.lang.Void
import io.swagger.v3.oas.annotations.parameters.RequestBody
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service

@CrossOrigin
@RestController
@Service
class ScoreboardAPI @Autowired constructor(private val scoreRepository: ScoreRepository) {

    private val log = LoggerFactory.getLogger(this.javaClass)

    @RequestMapping("/")
    fun showRules(): String {
        return "Hi there"
    }

    @RequestMapping("/rules")
    fun showRules2(): String {
        return "Hi there"
    }

    @PostMapping("/scoreboard/{gameId}/postName")
    fun postNameOnScoreboard(
        @PathVariable gameId: Long,
        @RequestBody @org.springframework.web.bind.annotation.RequestBody name: String
    ): ResponseEntity<Void> {
        val gameFinished = true //TODO
        val timestamp = LocalDateTime.now()
        val scoreBE = Score(gameId, name, 42.0, timestamp)
        scoreRepository.save(scoreBE)
        return ResponseEntity.ok().build()
    }
}